kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java

```diff
@@ -59,38 +66,48 @@ public KafkaConsumerInput( StepMeta stepMeta, StepDataInterface stepDataInterfac
    * @param stepDataInterface The data to initialize
    */
   @Override public boolean init( StepMetaInterface stepMetaInterface, StepDataInterface stepDataInterface ) {
-    KafkaConsumerInputMeta kafkaConsumerInputMeta = (KafkaConsumerInputMeta) stepMetaInterface;
-    KafkaConsumerInputData kafkaConsumerInputData = (KafkaConsumerInputData) stepDataInterface;
-
+    kafkaConsumerInputMeta = (KafkaConsumerInputMeta) stepMetaInterface;
+    kafkaConsumerInputData = (KafkaConsumerInputData) stepDataInterface;
     boolean superInit = super.init( kafkaConsumerInputMeta, kafkaConsumerInputData );
     if ( !superInit ) {
       logError( BaseMessages.getString( PKG, "KafkaConsumerInput.Error.InitFailed" ) );
       return false;
     }
 
+    if ( !kafkaConsumerInputMeta.getKafkaFactory().checkKafkaConnectionStatus(
+            kafkaConsumerInputMeta, variables, getLogChannel() ) ) {
+      return false;
+    }
+
     try {
       kafkaConsumerInputData.outputRowMeta = kafkaConsumerInputMeta.getRowMeta( getStepname(), this );
     } catch ( KettleStepException e ) {
       log.logError( e.getMessage(), e );
     }
 
-    Consumer consumer =
-      kafkaConsumerInputMeta.getKafkaFactory().consumer( kafkaConsumerInputMeta, this::environmentSubstitute,
-        kafkaConsumerInputMeta.getKeyField().getOutputType(),
-        kafkaConsumerInputMeta.getMessageField().getOutputType() );
+    this.prepareConsumer( kafkaConsumerInputMeta, kafkaConsumerInputData );
+
+    return true;
+  }
+
+  private void commitOffsets( Map.Entry<List<List<Object>>, Result> rowsAndResult ) {
+    ( (KafkaStreamSource) source ).commitOffsets( rowsAndResult.getKey() );
+  }
+
+  protected void prepareConsumer( KafkaConsumerInputMeta kafkaConsumerInputMeta,
+                                  KafkaConsumerInputData kafkaConsumerInputData ) {
+    Consumer consumer = kafkaConsumerInputMeta.getKafkaFactory().consumer( kafkaConsumerInputMeta,
+            this::environmentSubstitute, kafkaConsumerInputMeta.getKeyField().getOutputType(),
+            kafkaConsumerInputMeta.getMessageField().getOutputType() );
 
     Set<String> topics =
-      kafkaConsumerInputMeta.getTopics().stream().map( this::environmentSubstitute ).collect( Collectors.toSet() );
+            kafkaConsumerInputMeta.getTopics().stream().map( this::environmentSubstitute ).collect( Collectors.toSet() );
```

Comment:
Please check your code style settings and fix this and other whitespace changes in this PR.